### DISCUSSION TITLE
BepInEx and ValheimPlus not working

### DISCUSSION THREAD

**Discussion by ottobaer** [mbround18/valheim-docker#587] [Help!]
Hi, 

The server works if I start it vanilla, but if I want to use BepInEx or ValheimPlus I'm getting a lot of error messages and then it segfaults. 

Any idea where this could be coming from? 

```
ODIN][INFO]  - BepInEx detected! Switching to run with BepInEx...
[ODIN][INFO]  - BepInEx found! Setting up Environment...
[ODIN][ERROR] - Error: unable to set group
Found UnityPlayer, hooking into it instead
Mono path[0] = '/home/steam/valheim/valheim_server_Data/Managed'
Mono config path = '/home/steam/valheim/valheim_server_Data/MonoBleedingEdge/etc'
Got override: /home/steam/valheim/unstripped_corlib
Current root dir: /home/steam/valheim/valheim_server_Data/Managed
New root path: /home/steam/valheim/unstripped_corlib
Search path: /home/steam/valheim/unstripped_corlib:/home/steam/valheim/valheim_server_Data/Managed
Setting config paths; basedir: /home/steam/valheim; config: /home/steam/valheim/valheim_server.x86_64.config
Managed dir: /home/steam/valheim/valheim_server_Data/Managed
Fallback handler could not load library /home/steam/valheim/valheim_server_Data/Mono/<D0><E4>+u<83>U/mono/aot-cache/amd64/BepInEx.Prelo
``````
```

Unable to find method ClearSubsystems in [UnityEngine.SubsystemsModule.dll]UnityEngine.SubsystemManager
Unable to find method InitializeIntegratedSubsystem in [UnityEngine.SubsystemsModule.dll]UnityEngine.SubsystemManager
Unable to find method ReloadSubsystemsCompleted in [UnityEngine.SubsystemsModule.dll]UnityEngine.SubsystemManager
Unable to find method ReloadSubsystemsStarted in [UnityEngine.SubsystemsModule.dll]UnityEngine.SubsystemManager
Unable to find method ClearManagedDescriptors in [UnityEngine.SubsystemsModule.dll]UnityEngine.SubsystemsImplementation.SubsystemDescriptorStore
Unable to find method InitializeManagedDescriptor in [UnityEngine.SubsystemsModule.dll]UnityEngine.SubsystemsImplementation.SubsystemDescriptorStore
Caught fatal signal - signo:11 code:1 errno:0 addr:(nil)
Obtained 19 stack frames.
#0  0x007f04bac0d3c0 in funlockfile
#1  0x007f04ba2cd4d3 in mono_print_method_from_ip
#2  0x007f04ba43db18 in mono_perfcounter_foreach
#3  0x007f04ba43e9c5 in mono_runtime_invoke
#4  0x007f04bbc49b62 in scripting_method_invoke(ScriptingMethodPtr, ScriptingObjectPtr, ScriptingArguments&, ScriptingExceptionPtr*, bool)
#5  0x007f04bbc47c72 in ScriptingInvocation::Invoke(ScriptingExceptionPtr*, bool)
#6  0x007f04bb47c151 in Scripting::UnityEngine::SubsystemsImplementation::SubsystemDescriptorStoreProxy::ClearManagedDescriptors(ScriptingExceptionPtr*)
#7  0x007f04bb63cd95 in SubsystemManager::RebuildScriptingClassMap()
#8  0x007f04bbc2fbb9 in MonoManager::EndReloadAssembly(DomainReloadingData&, dynamic_bitset)
#9  0x007f04bbc2decf in MonoManager::ReloadAssembly()
#10 0x007f04bbc99a37 in AwakeFromLoadQueue::InvokePersistentManagerAwake(AwakeFromLoadQueue::Item*, unsigned int, AwakeFromLoadMode, bool)
#11 0x007f04bbc992c5 in AwakeFromLoadQueue::PersistentManagerAwakeFromLoad(ErrorsAndWarningsCapture*, bool)
#12 0x007f04bbc85337 in PersistentManager::LoadAndIntegrateAllPreallocatedObjects(PersistentManager::LockFlags)
#13 0x007f04bbc858eb in PersistentManager::ReadObject(int, AwakeFromLoadMode)
#14 0x007f04bb897834 in PlayerLoadGlobalManagers(char const*, char const*, unsigned int)
#15 0x007f04bbacef39 in PlayerInitEngineGraphics(bool)
#16 0x007f04bbca7ab3 in PlayerMain(int, char**)
#17 0x007f04baa2b0b3 in __libc_start_main
#18 0x00558373c9d029 in _start

```

**Comment by mbround18** [comment]
@ottobaer what version of the game and bepinex are you running? Most likely, if valheim updated then you will see this until they update as well.

**Reply by ottobaer** [reply]
Thank you, I haven't thought about that. 


This is what gets installed 
```

Valheim version:0.209.10
BepInEx 5.4.19.0 

```
From the changelog 

```

5.4.1901

    Updated to support Valheim 0.209.5

5.4.1900

    Updated to BepInEx 5.4.19 
```

So I guess there are two things here. 

The docker image is using version 5.4.1900 and the versions 5.x are going to be deprecated for the upcoming 6.x release of BepInEx. 

So that explains it. 

If I have time I will try to manually fetch the 6.x release with the docker-compose options, theoretically it should work I guess.

**Reply by ottobaer** [reply]
I just tried installing the latest BepInEx with odin - it will just start the vanilla game. I lack the knowledge on how to go on here I guess.

**Reply by mbround18** [reply]
@ottobaer thanks for pointing that out, i can take a look at updating the urls for 6.x

**Reply by mbround18** [reply]
They should all be pulling from latest, maybe the bepinex guys need to get their stuff updated? or they have new repos? 0.o ill investigate further

**Reply by spannerman79** [reply]
Thunderstore has updated BepInEx 5.4.1901 ( https://valheim.thunderstore.io/package/denikson/BepInExPack_Valheim/ ) but Github is still stuck at [5.4.1900](https://github.com/BepInEx/BepInEx/releases/tag/v5.4.19)

Thunderstore URL: https://gcdn.thunderstore.io/live/repository/packages/denikson-BepInExPack_Valheim-5.4.1901.zip

**Reply by spannerman79** [reply]
> If I have time I will try to manually fetch the 6.x release with the docker-compose options, theoretically it should work I guess.

@ottobaer the 6.0 releases are (currently) bleeding edge builds, obtained via https://builds.bepinex.dev/projects/bepinex_be

I wouldn't use those just yet unless you have backups and/or willing to accept data loss/data being FUBAR.

**Reply by ottobaer** [reply]
> 
the 6.0 releases are (currently) bleeding edge builds, obtained via https://builds.bepinex.dev/projects/bepinex_be

@spannerman79  - thanks for pointing that out. 

I must have missed that, my bad - I just saw that 5.x was deprecated so I assumed the 6.x is already stable.

**Reply by spannerman79** [reply]
> I must have missed that, my bad - I just saw that 5.x was deprecated so I assumed the 6.x is already stable.

Actually no, the `5.x` branch has entered [LTS](https://en.wikipedia.org/wiki/Long-term_support).  Not deprecated.  There is a difference.
